### ISSUE TITLE
20.1.0 ships graal_isolate(_dynamic).h file in both GraalVM and Native Image

### ISSUE THREAD

**Issue by lucaswerkmeister** [oracle/graal#2491]
**Describe the issue**
As I was upgrading the GraalVM AUR (Arch User Repository) packages to 20.1.0, I ran into an issue when installing the provisional Native Image 20.1.0 package locally:

```
error: failed to commit transaction (conflicting files)
native-image-jdk8-bin: /usr/lib/jvm/java-8-graalvm/jre/lib/amd64/graal_isolate.h exists in filesystem (owned by jdk8-graalvm-bin)
native-image-jdk8-bin: /usr/lib/jvm/java-8-graalvm/jre/lib/amd64/graal_isolate_dynamic.h exists in filesystem (owned by jdk8-graalvm-bin)
Errors occurred, no packages were upgraded.
```

And indeed those two header files are included both in the GraalVM tarball and in the Native Image component:

```console
$ tar tf graalvm-ce-java8-linux-amd64-20.1.0.tar.gz | grep graal_isolate
graalvm-ce-java8-20.1.0/jre/lib/amd64/graal_isolate_dynamic.h
graalvm-ce-java8-20.1.0/jre/lib/amd64/graal_isolate.h
graalvm-ce-java8-20.1.0/jre/lib/polyglot/graal_isolate_dynamic.h
graalvm-ce-java8-20.1.0/jre/lib/polyglot/graal_isolate.h
$ unzip -t native-image-installable-svm-java8-linux-amd64-20.1.0.jar | grep graal_isolate
    testing: jre/lib/amd64/graal_isolate_dynamic.h   OK
    testing: jre/lib/amd64/graal_isolate.h   OK
```

The Java 11 release has the same issue (though the file layout differs slightly):

```console
$ tar tf graalvm-ce-java11-linux-amd64-20.1.0.tar.gz | grep graal_isolate
graalvm-ce-java11-20.1.0/lib/graal_isolate_dynamic.h
graalvm-ce-java11-20.1.0/lib/graal_isolate.h
graalvm-ce-java11-20.1.0/lib/polyglot/graal_isolate_dynamic.h
graalvm-ce-java11-20.1.0/lib/polyglot/graal_isolate.h
$ unzip -t native-image-installable-svm-java11-linux-amd64-20.1.0.jar | grep graal_isolate
    testing: lib/graal_isolate_dynamic.h   OK
    testing: lib/graal_isolate.h      OK
```

And also on AArch64:

```console
$ tar tf graalvm-ce-java11-linux-aarch64-20.1.0.tar.gz | grep graal_isolate
graalvm-ce-java11-20.1.0/lib/graal_isolate.h
graalvm-ce-java11-20.1.0/lib/graal_isolate_dynamic.h
graalvm-ce-java11-20.1.0/lib/polyglot/graal_isolate.h
graalvm-ce-java11-20.1.0/lib/polyglot/graal_isolate_dynamic.h
$ unzip -t native-image-installable-svm-java11-linux-aarch64-20.1.0.jar | grep graal_isolate
    testing: lib/graal_isolate.h      OK
    testing: lib/graal_isolate_dynamic.h   OK
```

I’ve worked around this for the 20.1.0 release (the files are identical, so I just removed them from the Native Image package), but I assume this isn’t intentional, and in future releases those files should only be shipped once.

**Describe GraalVM and your environment:**
 - GraalVM version: **[CE 20.1.0](https://github.com/graalvm/graalvm-ce-builds/releases/tag/vm-20.1.0)**
 - JDK major version: **8** and **11**
 - OS: **Arch Linux**
 - Architecture: **x86_64** / **amd64** and **aarch64**

**Comment by lucaswerkmeister** [comment]
Issue still present in 20.2.0, it seems (though I only checked Java 8 + AMD64).

**Comment by lucaswerkmeister** [comment]
Issue still present in 22.1.0.

```console
$ curl -sL https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-22.1.0/graalvm-ce-java17-linux-amd64-22.1.0.tar.gz | tar tz | grep graal_isolate
graalvm-ce-java17-22.1.0/lib/graal_isolate.h
graalvm-ce-java17-22.1.0/lib/graal_isolate_dynamic.h
$ curl -sL https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-22.1.0/native-image-installable-svm-java17-linux-amd64-22.1.0.jar | bsdtar t | grep graal_isolate
lib/graal_isolate.h
lib/graal_isolate_dynamic.h
```

**Comment by lucaswerkmeister** [comment]
Seems to be resolved in 22.3.2 – neither package contains anything `graal_isolate` anymore:

```
$ curl -sL https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-22.3.2/graalvm-ce-java17-linux-amd64-22.3.2.tar.gz | tar tz | grep graal_isolate
$ curl -sL https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-22.3.2/native-image-installable-svm-java17-linux-amd64-22.3.2.jar | bsdtar t | grep graal_isolate
$
```
